connectors/spark/src/test/java/io/unitycatalog/spark/auth/AwsCredentialRenewalTest.java

```diff
@@ -0,0 +1,362 @@
+
+  private static final String CATALOG_NAME = "CredRenewalCatalog";
+  private static final String SCHEMA_NAME = "Default";
+  private static final String TABLE_NAME = String.format("%s.%s.demo", CATALOG_NAME, SCHEMA_NAME);
+  private static final String BUCKET_NAME = "test-bucket";
+  private static final String S3_SCHEME = "s3:";
+  private static final long DEFAULT_INTERVAL_MILLIS = 30_000L;
+
+  @TempDir private File dataDir;
+  private SparkSession session;
+  private SdkSchemaOperations schemaOperations;
+
+  @Override
+  protected CatalogOperations createCatalogOperations(ServerConfig config) {
+    return new SdkCatalogOperations(createApiClient(config));
+  }
+
+  @Override
+  protected void setUpProperties() {
+    super.setUpProperties();
+    serverProperties.put("s3.bucketPath.0", "s3://" + BUCKET_NAME);
+    serverProperties.put("s3.accessKey.0", "accessKey0");
+    serverProperties.put("s3.secretKey.0", "secretKey0");
+    serverProperties.put("s3.sessionToken.0", "sessionToken0");
+    serverProperties.put("s3.credentialsGenerator.0", CREDENTIALS_GENERATOR);
+  }
+
+  private SparkSession createSparkSession() {
+    String testCatalogKey = String.format("spark.sql.catalog.%s", CATALOG_NAME);
+    return SparkSession.builder()
+        .appName("test-credential-renewal")
+        .master("local[1]") // Make it single-threaded explicitly.
+        .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+        .config(
+            "spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
+        .config("spark.hadoop." + UCHadoopConf.UC_TEST_CLOCK_NAME, CLOCK_NAME)
+        .config("spark.hadoop." + UCHadoopConf.UC_RENEWAL_LEAD_TIME_KEY, 0L)
+        .config("spark.sql.shuffle.partitions", "1")
+        .config(testCatalogKey, UCSingleCatalog.class.getName())
+        .config(testCatalogKey + ".uri", serverConfig.getServerUrl())
+        .config(testCatalogKey + ".token", serverConfig.getAuthToken())
+        .config(testCatalogKey + ".warehouse", CATALOG_NAME)
+        .config(testCatalogKey + ".renewCredential.enabled", "true")
+        .config("fs.s3.impl", S3CredFileSystem.class.getName())
+        .getOrCreate();
+  }
+
+  private interface Callable {
+
+    void call() throws Exception;
+  }
+
+  private void callQuietly(Callable call) {
+    try {
+      call.call();
+    } catch (Exception e) {
+      // ignored.
+    }
+  }
+
+  @BeforeEach
+  public void beforeEach() throws Exception {
+    super.setUp();
+    session = createSparkSession();
+
+    // Initialize the catalog in unity catalog server.
+    catalogOperations.createCatalog(
+        new CreateCatalog().name(CATALOG_NAME).comment("Spark catalog"));
+
+    // Initialize the schema in unity catalog server.
+    schemaOperations = new SdkSchemaOperations(createApiClient(serverConfig));
+    schemaOperations.createSchema(new CreateSchema().name(SCHEMA_NAME).catalogName(CATALOG_NAME));
```

Comment:
cant you reuse the catalog + schema across tests with beforeAll? this is adding latency unnecessarily

Reply:
There are other test cases which depend on this `BaseCRUDTest` feature ( I mean the `beforeEach` level catalog & schema initialization and destory ) ,  if  we plan to change then maybe we should change those test cases firstly.